services/select.js

```diff
@@ -208,193 +167,250 @@ function addIframeOverlays(el) {
-  });
-
-  if (!componentBar.querySelector('.menu-toggle')) {
-    componentBar.appendChild(el);
+function addDragHandler(el, parent) {
+  if (parent.isComponentList) {
+    el.classList.add('drag');
   }
 }
 
 /**
  * add component bar (with click events)
- * @param {Element} componentEl   An element that has a ref.
+ * @param {Element} el component element
  * @param {object} options
  * @param {string} options.ref
  * @param {string} options.path
  * @param {object} options.data
  * @returns {Element}
  */
-function handler(componentEl, options) {
+function handler(el, options) {
   var name = references.getComponentNameFromReference(options.ref),
-    tpl = `
-    <aside class="component-bar">
-      <span class="label selected-label" title="${label(name)}">
-        <span class="selected-label-inner">${label(name)}</span>
-      </span>
-    </aside>
-    `,
-    componentBar = dom.create(tpl);
-
-  // Add options to the component bar.
-  addSettingsOption(componentBar, options.data, options.ref);
-  addParentOptions(componentBar, componentEl, options.ref);
-
-  // add events to the component itself
-  // when the component is clicked, it should be selected
-  componentEl.addEventListener('click', componentClickHandler.bind(null, componentEl));
-
-  // make sure components are relatively positioned
-  componentEl.classList.add('component-bar-wrapper');
-  dom.prependChild(componentEl, componentBar); // prepended, so parent components are behind child components
-  // add an iframe-overlay to iframes so we can click on components with them
-  addIframeOverlays(componentEl);
-  return componentEl;
+    selector = tpl.get('.component-selector-template');
+
+  // resolve parent info. if there are no parents this resolves to empty object
+  return getParentInfo(el).then(function (parent) {
+    // add options to the component selector
+    // set component label
+    addLabel(selector, name);
+    // if parent, unhide + add handler
+    addParentHandler(selector, parent);
+
+    // if settings, unhide + add handler
+    addSettingsHandler(selector, options);
+    // if delete, unhide + add handler
+    addDeleteHandler(selector, parent, el, options);
+
+    // if add, unhide bottom
+    // note: more options might exist in the bottom menu in the future
+    unhideBottomMenu(selector, parent);
+    // if add, unhide + add handler
+    addAddHandler(selector, parent, options);
+
+    // if drag, add class
+    // note: this adds a class to the component itself,
+    // not the selector
+    addDragHandler(el, parent);
+
+    // add iframe overlays
+    addIframeOverlays(el);
+
+    // add events to the component itself
+    // when the component is clicked, it should be selected
+    el.addEventListener('click', componentClickHandler.bind(null, el));
+
+    // make sure components are relatively positioned
+    el.classList.add('component-selector-wrapper');
+
+    // add selector to the component el
+    dom.prependChild(el, selector);
```

Comment:
todo: explain why (add selector behind the rest of the components' elements, so you can access them)